File: app/src/main/java/com/deniscerri/ytdlnis/service/IDownloaderListener.java
package com.deniscerri.ytdlnis.service;

public interface IDownloaderListener {
    void onDownloadStart(DownloadInfo downloadInfo);
    void onDownloadProgress(DownloadInfo downloadInfo);
    void onDownloadError(DownloadInfo downloadInfo);
    void onDownloadEnd(DownloadInfo downloadInfo);
    void onDownloadCancel(DownloadInfo downloadInfo);
    void onDownloadCancelAll(DownloadInfo downloadInfo);
    void onDownloadServiceEnd();
}

File: app/src/main/java/com/deniscerri/ytdlnis/service/IDownloaderListener.kt
package com.deniscerri.ytdlnis.service

interface IDownloaderListener {
    fun onDownloadStart(downloadInfo: DownloadInfo?)
    fun onDownloadProgress(downloadInfo: DownloadInfo?)
    fun onDownloadError(downloadInfo: DownloadInfo?)
    fun onDownloadEnd(downloadInfo: DownloadInfo?)
    fun onDownloadCancel(downloadInfo: DownloadInfo?)
    fun onDownloadCancelAll(downloadInfo: DownloadInfo?)
    fun onDownloadServiceEnd()
}